dbresolver.go

```diff
@@ -295,17 +295,35 @@ func hydrateConnections(db *sql.DB, conns, workers int) error {
 	return nil
 }
 
-func (dr *DBResolver) HydrateConnections(conns, workers int) error {
+func (dr *DBResolver) HydrateConnections(conns, workers int) (int, error) {
+	var opened int
 	globalResolver := dr.global
 	for _, s := range globalResolver.sources {
 		if db, ok := s.ConnPool.(*sql.DB); ok {
-			hydrateConnections(db, conns, workers)
+			diff := conns - db.Stats().OpenConnections + 1
```

Comment:
why +1 here and below?

Reply:
When I tested this on a short 5s interval, `OpenConnections` was always reporting 1 less connection than the target, which I didn't quite understand. Given we're not hydrating on that short an interval I suppose this +1 is less relevant